eth/backend.go

```diff
@@ -196,12 +215,18 @@ func New(stack *node.Node, config *ethconfig.Config) (*Ethereum, error) {
 			Preimages:           config.Preimages,
 		}
 	)
+	if readOnly {
+		cacheConfig.SnapshotLimit = 0
+		cacheConfig.TrieDirtyDisabled = true
+		cacheConfig.TrieCleanJournal = ""
+		config.TxPool.Journal = ""
+	}
```

Comment:
These params should have been set appropriately in `makeConfigNode` from `readOnlyGeth/config.go`, rather than changed after-the-fact here.